### PR TITLE
PM-11176: Update generator to use segmented control

### PR DESCRIPTION
**Pull request by david-livefront**
## 🎟️ Tracking

[PM-11176](https://bitwarden.atlassian.net/browse/PM-11176)

## 📔 Objective

This PR updates the Generator Screen to use a Segmented Control for top level navigation.

## 📸 Screenshots

| Before | After |
| --- | --- |
|  |  |

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

[PM-11176]: https://bitwarden.atlassian.net/browse/PM-11176?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/ui/platform/components/segment/BitwardenSegmentedButton.kt [hunk 1]
```diff
@@ -46,6 +46,7 @@ fun BitwardenSegmentedButton(
         ) {
             options.forEachIndexed { index, option ->
                 SegmentedButton(
+                    enabled = option.isEnabled,
```
Comment (dseverns-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/platform/components/segment/BitwardenSegmentedButton.kt:49]: 👍

File: app/src/main/java/com/x8bit/bitwarden/ui/tools/feature/generator/GeneratorViewModel.kt [hunk 2]
```diff
@@ -308,72 +304,91 @@ class GeneratorViewModel @Inject constructor(
...
-                )
+
+            GeneratorState.PasscodePolicyOverride.PASSPHRASE -> {
+                mutableStateFlow.update {
+                    it.copy(passcodePolicyOverride = GeneratorState.PasscodePolicyOverride.PASSPHRASE)
+                }
+                GeneratorState.MainType.Passphrase()
+            }
+
+            null -> {
```
Comment (dseverns-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/tools/feature/generator/GeneratorViewModel.kt:335]: 👍  nice

**Review by dseverns-livefront — APPROVED** [review]

**Comment by david-livefront** [comment]
Thanks @dseverns-livefront

**Pull request merged**